climakitae/threshold_tools.py

```diff
@@ -42,121 +46,143 @@ def get_ams(da, extremes_type="max"):
         ams.attrs["block size"] = "1 year"
         ams.attrs["timeseries type"] = "annual max series"
 
-    # if extremes_type == 'min':
-    #     ams = da.resample(time='A').min(keep_attrs = True)
-    #     ams.attrs['extreme value extraction method'] = 'block maxima'
-    #     ams.attrs['extremes type'] = 'minima'
-    #     ams.attrs['block size'] = '1 year'
-    #     ams.attrs['timeseries type'] = 'annual min series'
     return ams
 
 
-def get_lmom_distr(distr):
-    """
-    Returns corresponding l-moments distribution function from selected
+def get_distr_func(distr):
+    """Function that sets the scipy distribution object
+
+    Sets corresponding distribution function from selected
     distribution name.
+
+    Parameters
+    ----------
+    distr: str
+        name of distribution to use
+
+    Returns
+    -------
+    scipy.stats
     """
 
     distrs = ["gev", "gumbel", "weibull", "pearson3", "genpareto"]
 
     if distr == "gev":
-        lmom_distr = ldistr.gev
+        distr_func = stats.genextreme
     elif distr == "gumbel":
-        lmom_distr = ldistr.gum
+        distr_func = stats.gumbel_r
     elif distr == "weibull":
-        lmom_distr = ldistr.wei
+        distr_func = stats.weibull_min
     elif distr == "pearson3":
-        lmom_distr = ldistr.pe3
+        distr_func = stats.pearson3
     elif distr == "genpareto":
-        lmom_distr = ldistr.gpa
+        distr_func = stats.genpareto
     else:
         raise ValueError(
-            "invalid distr type. expected one of the following: %s" % distrs
+            "invalid distribution type. expected one of the following: %s" % distrs
         )
 
-    return lmom_distr
+    return distr_func
 
 
-def get_fitted_distr(ams, distr, lmom_distr):
-    """
-    Returns fitted l-moments distribution function from l-moments.
+def get_fitted_distr(ams, distr, distr_func):
+    """Function for fitting data to distribution function
+
+    Takes data array and fits it to distribution function.
+
+    Parameters
+    ----------
+    ams: xarray.DataArray
+    distr: str
+    distr_func: scipy.stats
+
+    Returns
+    -------
+    parameters: dict
+        dictionary of distribution function parameters
+    fitted_distr: scipy.rv_frozen
+        frozen fitted distribution
     """
 
-    lmoments = lmom_distr.lmom_fit(ams)
+    def get_parameters(p, p_values):
+        return {p[i]: p_values[i] for i, _ in enumerate(p_values)}
+
+    parameters = None
+    fitted_distr = None
+
+    p_values = distr_func.fit(ams)
 
     if distr == "gev":
-        fitted_distr = stats.genextreme(**lmoments)
+        p = ("c", "loc", "scale")
```

Comment:
if my above suggestion makes sense, I will also just rename each of these `p` lists as `p_names` for clarity!